src/pages/workspaces/workspace/Notebooks.js

```diff
@@ -51,93 +53,168 @@ export default hh(class WorkspaceNotebooks extends Component {
-                    overflow: 'hidden'
-                  }
-                }, creator)
-              },
-              {
-                title: 'Delete', key: 'delete',
-                render: ({ clusterName, googleProject, status }) => {
-                  if (status !== 'Deleting') {
-                    return link({
-                      onClick: () => {
-                        Leo.clusterDelete(googleProject, clusterName,
-                          () => this.loadClusters(),
-                          deletionFail => window.alert(`Couldn't delete cluster: ${deletionFail}`)
-                        )
-                      },
-                      title: `Delete cluster ${clusterName}`
-                    }, [icon('trash', { style: { margin: 'auto', display: 'block' } })])
                   }
                 }
-              }
-            ]
-          }
-        })
-      ])
-    )
+              ]
+            }
+          }),
+          div({
+            style: { fontSize: 16, fontWeight: 500, color: Style.colors.title, marginTop: '2rem' }
+          }, 'NOTEBOOKS'),
+          Utils.cond(
+            [notebooksFailure, () => `Couldn't load cluster list: ${notebooksFailure}`],
+            [!notebooks, spinner],
+            () => div({ style: { display: 'flex' } },
+              _.map(notebooks, ({ name, updated }) => a({
+                  target: '_blank',
+                  href: notebookAccess[name] ?
+                    `${_.first(clusters).clusterUrl}/notebooks/${workspace.name}/${name.slice(10)}` :
+                    undefined,
+                  style: _.defaults({
+                    width: 200, height: 250,
+                    margin: '1.25rem', boxSizing: 'border-box',
+                    textDecoration: 'none',
+                    cursor: !notebookAccess[name] ? 'not-allowed' : undefined,
+                    display: 'flex', flexDirection: 'column',
+                    justifyContent: 'space-between',
+                    color: Style.colors.text
+                  }, Style.elements.card)
+                },
+                [
+                  div({
+                    style: _.defaults(notebookAccess[name] ? {} : { color: Style.colors.disabled },
+                      Style.elements.cardTitle)
+                  }, name.slice(10, -6)), // ignores 'notebooks/' and the .ipynb suffix
+                  icon('jupyterIcon',
+                    { style: { height: 125, width: 'auto', color: Style.colors.background } }),
+                  div({ style: { display: 'flex', alignItems: 'flex-end' } }, [
+                    div({ style: { fontSize: '0.8rem', flexGrow: 1 } }, [
+                      'Last changed:',
+                      div({}, Utils.makePrettyDate(updated))
+                    ]),
+                    Utils.cond(
+                      [notebookAccess[name] === false, () => icon('times', { title: 'Error' })],
```

Comment:
I liked this when I saw it, but even better might be to link it to the localization.log in the cluster.

Reply:
Once we have the synchronous version of this call, this will be moot.